token/metadata.go

```diff
@@ -43,124 +48,135 @@ func (m *Metadata) SpentTokenID() []*token.ID {
+			Inputs:       issue.Inputs,
+			Outputs:      nil,
 			ExtraSigners: issue.ExtraSigners,
 		}
 
-		for i, receivedAuditInfo := range issue.ReceiversAuditInfos {
-			// If the receiver has the given enrollment ID, add it
-			recipientEID, err := m.WalletService.GetEnrollmentID(issue.Receivers[i], receivedAuditInfo)
-			if err != nil {
-				return nil, errors.Wrap(err, "failed getting enrollment ID")
+		counter := 0
+		for _, output := range issue.Outputs {
+			found := false
+			for _, receiver := range output.Receivers {
+				// If the receiver has the given enrollment ID, add it
+				recipientEID, err := m.WalletService.GetEnrollmentID(receiver.Identity, receiver.AuditInfo)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed getting enrollment ID")
+				}
+				if eIDSet.Contains(recipientEID) {
+					found = true
+					break
+				}
 			}
-			var outputMetadata []byte
-			receivers := issue.Receivers[i]
-			var receiverAuditInfos []byte
-
-			if search(eIDs, recipientEID) != -1 {
-				outputMetadata = issue.OutputsMetadata[i]
-				receiverAuditInfos = issue.ReceiversAuditInfos[i]
+			if found {
+				clone.Outputs = append(clone.Outputs, output)
+				counter++
 			} else {
-				m.Logger.Debugf("skipping issue for [%s]", recipientEID)
+				clone.Outputs = append(clone.Outputs, nil)
 			}
-
-			issueRes.OutputsMetadata = append(issueRes.OutputsMetadata, outputMetadata)
-			issueRes.Receivers = append(issueRes.Receivers, receivers)
-			issueRes.ReceiversAuditInfos = append(issueRes.ReceiversAuditInfos, receiverAuditInfos)
 		}
 
-		res.TokenRequestMetadata.Issues = append(res.TokenRequestMetadata.Issues, issueRes)
+		m.Logger.Debugf("keeping issue with [%d] out of [%d] outputs", counter, len(issue.Outputs))
+		clonedMetadata.TokenRequestMetadata.Issues = append(clonedMetadata.TokenRequestMetadata.Issues, clone)
 	}
 
 	// filter transfers
 	for _, transfer := range m.TokenRequestMetadata.Transfers {
-		transferRes := driver.TransferMetadata{
+		clone := &driver.TransferMetadata{
+			Inputs:       nil,
+			Outputs:      nil,
 			ExtraSigners: transfer.ExtraSigners,
 		}
 
 		// Filter outputs
 		// if the receiver has the given enrollment ID, add it. Otherwise, add empty entries
 		skip := true
-		for i, auditInfo := range transfer.ReceiverAuditInfos {
-			recipientEID, err := m.WalletService.GetEnrollmentID(transfer.Receivers[i], auditInfo)
-			if err != nil {
-				return nil, errors.Wrap(err, "failed getting enrollment ID")
+		counter := 0
+		for _, output := range transfer.Outputs {
+			found := false
+			for _, receiver := range output.Receivers {
+				// If the receiver has the given enrollment ID, add it
+				recipientEID, err := m.WalletService.GetEnrollmentID(receiver.Identity, receiver.AuditInfo)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed getting enrollment ID")
+				}
+				if eIDSet.Contains(recipientEID) {
+					logger.Debugf("eid [%s] found in list [%v]", recipientEID, eIDs)
+					skip = false
```

Comment:
The variable `skip` can be replaced with `counter == 0`

Reply:
okay